presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithExchangeMaterialization.java

```diff
@@ -171,6 +171,18 @@ private void testBucketedByHiveUnsupportedTypeForTemporaryTable(
         assertUpdate("DROP TABLE IF EXISTS test_materialize_bucket_by_non_hive_types");
     }
 
+    // We need to disable optimize_constant_grouping_keys so Optimizer does NOT optimize
```

Comment:
can you explain what's going on here and also why this changed with migration to RowExpressions?

Reply:
Lol yeah. Look at testRemoveConstantBeforeGroupBy - which is a new test that I added.

Moving from SimplifyExpressions -> SimplifyRowExpressions, makes this query shape being optimized properly. It pulls the constant above a group by correctly, which it wasn't doing it before. Which led to this test failing(because it expects particular types in exchanges). So i disabled this optimization here - and added another test for new behaviour.

Reply:
Did we consider looking in making SimplifyRowExpressions behave same as SimplifyExpressions?

Reply:
Ignore my last comment, I read incorrectly, I thought SimplifyRowExpressions was producing non optimized query.